ip.go

```diff
@@ -14,8 +14,8 @@ var (
 	// IP6Loopback is the ip6 loopback multiaddr
 	IP6Loopback = ma.StringCast("/ip6/::1")
 
-	// IP6LinkLocalLoopback is the ip6 link-local loopback multiaddr
-	IP6LinkLocalLoopback = ma.StringCast("/ip6/fe80::1")
+	// IP4MappedIP6Loopback is the IPv4 Mapped IPv6 loopback address.
+	IP4MappedIP6Loopback = ma.StringCast("/ip6/127.0.0.1")
```

Comment:
@lgierth Go accepts this as an alternative to `::ffff:127.0.0.1`. Are we OK with that?

Reply:
This seems to be go-specific and will be a source of incompatibilities between various implementations of multiaddr. Regardless, I feel like multiaddr should stick to the standards. So, in my opinion, not only should we not be OK with that, but we should also change go-multiaddrI so it refuses this syntax.

Reply:
You're probably right. I figured this could be useful as it makes it clear that we're using IPv4 mapped IPv6 but that *will* make it harder to parse so we should just go with what's simple.